conferences/ticketing_processing.go

```diff
@@ -0,0 +1,145 @@
+package conferences
+
+import (
+	"context"
+	"fmt"
+
+	"encore.dev/storage/sqldb"
+	uuid "github.com/satori/go.uuid"
+)
+
+// claimSlots claims N slots for an attendee.
+func claimSlots(ctx context.Context, attendee *Attendee, slots ...ConferenceSlot) ([]SlotClaim, error) {
+	tx, err := sqldb.Begin(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("beginning atomic operation: %w", err)
+	}
+	var claims = make([]SlotClaim, 0, len(slots))
+	for i := range slots {
+		slot := slots[i]
+		sc := &SlotClaim{
+			ConferenceSlot: &slot,
+			TicketID:       uuid.NewV4().String(),
+		}
+		sc, err = createSlotClaim(ctx, tx, sc)
+		if err != nil {
+			if atomicErr := sqldb.Rollback(tx); atomicErr != nil {
+				err = fmt.Errorf("%w (also cancelling atomic operation: %v)", err, atomicErr)
+			}
```

Comment:
This is not a named error, it is annotating the error we are about to return as we want to return both the root error and the rollback one if any (which likely means db is failing or we fail to connect to it)

Reply:
this implies a named return, I am not comfortable with it because these tend to have people get confused and sometimes, I prefer the explicitness of doing it before each return that implies rollback.
I do acknowledge though that both commit and rollback could be done in this way and I do not think its inherently bad but it is a pattern I have found to cause human error.